SalesforceFoundation/EDA
# [EDA] - Reciprocal Method Picklist Translation Fix

# Critical Changes

# Changes
We've created custom labels for List Setting and Value Inversion. These custom labels are referenced by the Reciprocal Method dropdown values under EDA Settings | Relationships | Settings so that those values can now be translated. The Reciprocal Method functionality hasn't changed.

# Issues Closed
Closes #1342 

# New Metadata
Custom Labels:

* stgReciprocalMethodListSetting
* stgReciprocalMethodValueInversion
# Deleted Metadata

# Testing Notes
Case A: Testing in English:

1) On load the default value should be 'List Setting' by default in EDA settings->Relationships->Settings
2) On edit, only 2 values should be visible 'List Setting'  and 'Value Inversion' . 'None' should never be an option.
3) The value in 'Reciprocal Method' field in the custom setting - 'Hierarchy Settings' (Setup -> Custom Settings -> Hierarchy Settings -> Manage) should be 'List Setting'
4) Changing the value in EDA Settings to 'Value Inversion' should change the value in Reciprocal Method' field in the custom setting to 'Value Inversion' as well.
5) Logging out/refreshing should preserve the values in both EDA settings UI and Custom Settings (Hierarchy Settings)

Case B: Testing in Spanish:
1) Add translations to the custom labels in Spanish, Setup -> Custom Label ->  stgReciprocalMethodListSetting -> Under translations, click New -> Language 'Spanish (Mexico)' say, 'Configuración de lista' and for the custom label stgReciprocalMethodValueInversion to  'Inversión de valor'.
2) Switch the language on your user to Spanish, Setup -> User -> User User -> Language -> 'Spanish'
3) Go to EDA settings->Relationships->Settings, the value for the 'Reciprocal Method' should now be 'Configuración de lista' and 'Inversión de valor' on edit screen. 'None' or its translated value should never be an option.
4) The values in 'Hierarchy Settings' (Setup -> Custom Settings -> Hierarchy Settings -> Manage) should still be in English. If the value in EDA settings is 'Configuración de lista' then Hierarchy setting should reflect 'List Settings'. If it is 'Inversión de valor' then the value in Hierarchy settings should be 'Value Inversion' 
5) Custom Settings should always have values in English corresponding to translated values in Spanish
6) Logging out/refreshing should preserve the values in both EDA settings UI and Custom Settings (Hierarchy Settings)

Other Notes:
This required an update to the CMP_Picklist_Dropdown Lighitng component, which is is utilized by the following picklists in EDA settings.
Accounts and Contacts tab: 
1) Select a field for Preferred Phone to copy....
2) Default Contact Language Fluency
On Affiliations -> Settings
1) Status for Affiliations Not Deleted
2) Role for Created Affiliations
3) Status for Created Affiliations
All the above 5 should have 'none' as an option.
--
